lib/network/kmip/ClusterClient.ts

```diff
@@ -270,7 +270,23 @@ export default class ClusterClient implements KMSInterface {
     }
 
     healthcheck(...args: Parameters<Required<KMSInterface>['healthcheck']>) {
-        // for now check health of every member
-        this.clusterHealthcheck.apply(this, args);
+        // healthcheck follow round robin with retries instead of broadcast
+        const originalCallback = args.pop() as ArrayLast<typeof args>;
+        const poppedArgs = args as unknown as ArrayPopped<typeof args>;
+        const logger = args[args.length - 1] as ArrayLast<typeof poppedArgs>;
+
+        if (this.clients.length === 0) {
+            logger.warn('kmip cluster has no healthy hosts');
+            return originalCallback(errorInstances.InternalError.customizeDescription(
+                kmipMsg('Healthcheck', '', `no healthy host in the cluster`)));
```

Comment:
Maybe return the same logged message to avoid surprises (and you can replace backticks by simple quotes):

```suggestion
                kmipMsg('Healthcheck', '', 'kmip cluster has no healthy hosts')));
```

Reply:
I've reworded in every methods